core/modules/views/css/views.css

```diff
@@ -56,3 +56,7 @@
 .view .progress-disabled {
   float: none;
 }
+
+.views-grid .views-row {
+  display: flex;
```

Comment:
Another thing with this, this will need a fallback for browsers that don't support flexbox. I can help with that as well.